minvws/nl-kat-coordination
# Basic audit trails via logging

### Changes

This PR implements some more audit trails via logging:
- Adding, creating and deleting objects
- Plugins
  - Change plugin state
  - Update plugin settings
- Raw data has been uploaded
- Login, logout and failed logins
- Organization creation and deletion

Most of the other logging related to models have been implemented in #3314 and merged into this branch.

Also implemented basic event codes for specific ORM record trails. This is a very basic implementation to get things started. I also considered using a tuple for the event codes (an element for each CRUDE action), but since there's probably still room for improvement and development around the audit trails I used a dict for now for flexibility.

Ideally we'd use more descriptive codes (using alphanumerics) when possible. In both cases, whether we use alphanumeric or numeric codes, we should work towards automatically labeling them as much as possible.

### Issue link

Partially closes #3043 

### QA notes

You should see more logging on certain actions, such as creating an object or enabling a plugin. On certain model changes (such as an organization or organization tags), you should also see an `event_code` variable in the log record.


---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
